### PR TITLE
Brake Pedal Position and Brake Pedal Light Logic

### PR DESCRIPTION
**Pull request by jabrap1**

### CODE REVIEW

File: firmware/shared/controls/brake_pedal_lights_test.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+#include <iostream>
+
+#include "testing.h"
+
+int main() {
+    using namespace ctrl;
+
+    // Test Case 1: Brake Pedal Map Edge Cases
+    {
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights_test.cc:11]: why do you have extra scoping? call functions directly in ASSERT

File: firmware/shared/controls/brake_pedal_lights_test.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        result = brakePedalMap(100.0f);
+        ASSERT_CLOSE(result, 100.0f);
+
+        result = brakePedalMap(50.0f);
+        ASSERT_CLOSE(result, 50.0f);
+    }
+
+    // Test Case 2: Brake Pedal Map with Interpolated Value
+    {
+        float result = brakePedalMap(12.5f);
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights_test.cc:24]: inline

File: firmware/shared/controls/brake_pedal_lights.h [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include <iostream>
+#include <tuple>
+
+#include "../util/mappers/lookup_table.h"
+
+namespace ctrl {
+
+// Brake Pedal Map Function
+template <typename T>
+T brakePedalMap(T input) {
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:12]: delete this function

File: firmware/shared/controls/brake_pedal_lights.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    T brakePedalPos;
+
+    // Brake Error Handling
+    if (rearBrakePedalError || frontBrakePedalError) {
+        brakePedalPos = 0;
+    } else {
+        T brakeInput = std::max(frontBrakePedalPos, rearBrakePedalPos);
+
+        // Brake Pedal Mapping
+        brakePedalPos = brakePedalMap(brakeInput);
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:39]: brake = input

**Review by BlakeFreer — CHANGES_REQUESTED** [review]

File: firmware/shared/controls/brake_pedal_lights.h [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    constexpr int size = (sizeof(lookupTableData) / sizeof(lookupTableData[0]));
+
+    shared::util::LookupTable<size> brakePedalMap(lookupTableData);
+
+    return brakePedalMap.Evaluate(input);
+}
+
+template <typename T>
+std::tuple<T, bool> brake_pedal_lights(bool rearBrakePedalError,
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:26]: split this into DetectBrakeLight and CalculateBrakePedalPosition

File: firmware/shared/controls/brake_pedal_lights.h [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                                       bool frontBrakePedalError,
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:27]: float front_brake_pedal_error

Comment (BlakeFreer) [reply firmware/shared/controls/brake_pedal_lights.h:27]: https://google.github.io/styleguide/cppguide.html#Function_Names

**Review by BlakeFreer — CHANGES_REQUESTED** [review]

File: firmware/shared/controls/brake_pedal_lights.h [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+namespace ctrl {
+
+template <typename T>
+T CalculateBrakePedalPosition(bool rear_brake_pedal_error,
+                              bool front_brake_pedal_error,
+                              T rear_brake_pedal_position,
+                              T front_brake_pedal_position) {
+    // Brake Error Handling
+    if (rear_brake_pedal_error || front_brake_pedal_error) {
+        return T{};  // removed the initalized variable because i realized i can
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:14]: This is a bad way to initialize a variable to 0. In not even sure if it's guaranteed to work by the c++ standard. Use `static_cast<T>(0)` instead

**Review by BlakeFreer — CHANGES_REQUESTED** [review]
Tests are good and complete

File: firmware/shared/controls/brake_pedal_lights.h [hunk 8]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <algorithm>
+#include <iostream>
```
Comment (BlakeFreer) [inline firmware/shared/controls/brake_pedal_lights.h:3]: unused  import

**Review by BlakeFreer — CHANGES_REQUESTED** [review]

**Review by BlakeFreer — APPROVED** [review]
good job
